dashboard/src/components/HeadManager/HeadManager.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { Helmet } from "react-helmet";
+import { useIntl } from "react-intl";
+
+interface IHeadManagerProps {
+  theme: SupportedThemes;
+  children: React.ReactNode;
+}
+
+export enum SupportedThemes {
+  dark = "dark",
+  light = "light",
+}
+
+export function getThemeFile(theme: SupportedThemes) {
+  const lightThemeFile = "./clr-ui.min.css";
+  const darkThemeFile = "./clr-ui-dark.min.css";
+  switch (theme) {
+    case SupportedThemes.light:
+      return lightThemeFile;
+    case SupportedThemes.dark:
+      return darkThemeFile;
+    default:
+      return lightThemeFile;
+  }
+}
+export default function HeadManager({ theme, children }: IHeadManagerProps) {
+  const intl = useIntl();
+
+  document.body.setAttribute("cds-theme", theme); // sets the initial cds theme
+  localStorage.setItem("theme", theme); // persist the initial theme decision
```

Comment:
I'm not getting your point here. Do you mean if this value can be overridden later?
If so, yes. The menu.tsx does changes both the `cds-theme` attribute and the `localStorage` value.

Reply:
no, my question was that if `theme` is the default in the browser (e.g. light) and the user has previously set it to dark, if this would change it back to light. Mostly I don't know where `theme` is coming from here, to be safe, I would do:

```
if (localStorage.getItem("theme") == "") {
  localStorage.setItem("theme", theme)
}
```

(if that is what it means)

Reply:
Ahh, got it. The `theme` value is something we own. The logic to initialize the value is:


```javascript
  const userPreferredTheme = localStorage.getItem("theme");
  const browserPreferredTheme =
    window.matchMedia && window.matchMedia("(prefers-color-scheme: dark)").matches
      ? SupportedThemes.dark
      : SupportedThemes.light;
  return (userPreferredTheme as SupportedThemes) || browserPreferredTheme || SupportedThemes.light;
```

So, there is no problem with setting the `theme` property. Once the user manually sets the preference, this value will override the browser's preference.
That is, the browser's preference is just used for the first load. Then it will be stored and only can be changed through the toggle button.

Reply:
so as far as I can tell, this line is not needed? It's just weird to set the value here in the browser when here we are not getting the default from the browser neither getting input from the user.

Reply:
Well, we simply can default it to `light` theme. However, this initial browser preference allows us to initialize the look&feel during the very first access to Kubeapps.
So, are you suggesting just remove the `browserPreferredTheme` ?